### PR TITLE
completed sgt-back-end - Jonathan Chiang

### PR DESCRIPTION
**Pull request by J0N-C**
GET:
sgt-back-end-get

POST:
sgt-back-end-post

PUT:
sgt-back-end-put

DELETE:
sgt-back-end-delete

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  const newEntry = req.body;
+  for (const property in newEntry) {
+    if (newEntry[property] === undefined || newEntry[property].length === 0) {
+      return res.status(400).json({ error: `${property} is a required field.` });
+    }
+  }
+  newEntry.score = Number(newEntry.score);
+  if (!Number.isInteger(newEntry.score) || newEntry.score > 100 || newEntry.score < 0) {
+    return res.status(400).json({ error: 'score must be an integer between 0 - 100' });
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: What happens if the request body only has a score and nothing else?

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({ error: 'gradeId must be a positive integer' });
+  }
+  const newEntry = req.body;
+  for (const property in newEntry) {
+    if (newEntry[property] === undefined || newEntry[property].length === 0) {
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: What happens if `newEntry[property]` isn't a string?

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 1 above)
Comment (J0N-C) [reply sgt-back-end/index.js]: Adding screenshot showing attempt to post with score only, the first if-statement checking for required fields still returns 400 error.
sgt-back-end-score-only-post

File: sgt-back-end/index.js (hunk 2 above)
Comment (J0N-C) [reply sgt-back-end/index.js]: Showing validation of score after splitting the for-in loop to separate validations, both post and put share the same one so I only did post here.

sgt-back-end-post-rev

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**